controllers/pvc_storageclass_test.go

```diff
@@ -139,19 +140,20 @@ var _ = Describe("The default StorageClass has not been configured", func() {
 			}()
 
 			By("Expecting the PVC has to be unchanged")
-			Consistently(func() corev1.PersistentVolumeClaim {
+			Consistently(func() string {
 				got := corev1.PersistentVolumeClaim{}
 				err := k8sClient.Get(ctx, client.ObjectKeyFromObject(&pvc), &got)
 				if err != nil {
-					return corev1.PersistentVolumeClaim{}
+					return "kube get error"
 				}
-				return got
-			}, storageClassTimeout, storageClassInterval).Should(Equal(pvc))
+
+				return got.Annotations[provisioner.StorageClassUUIDAnnotationKey]
+			}, storageClassTimeout, storageClassInterval).Should(BeEmpty())
```

Comment:
Ensuring we don't change anything else in the PVC sounds better since we are taking the whole object and changing only a part of it.

Reply:
I spoke with @mhmxs about this and he explained why we can't do this - the pre-create PVC won't have metadata like the PVC UID so a compare of the whole object will always fail.  It passed before as I missed that because I was passing in pvc as a reference, the create call was updating it with the post-create object.  The test always passes.

So I think it's ok to only check that we don't add the annotation.  But the test case should be renamed to explain this, perhaps with a comment why we don't compare the whole object.

Reply:
Thank you for your describe. I haven't commented the code, because other scenarios also test annotation only. I left some comment.

Reply:
thanks!